Specs/Core/Matrix3Spec.js

```diff
@@ -148,6 +148,24 @@ defineSuite([
         expect(returnedResult).toEqualEpsilon(expected, CesiumMath.EPSILON15);
     });
 
+    it('fromHeadingPitchRoll computed correctly', function() {
+        var sPi = Math.sin(CesiumMath.PI);
+        var cPi = Math.cos(CesiumMath.PI);
+        var sPiOver2 = Math.sin(CesiumMath.PI_OVER_TWO);
+        var cPiOver2 = Math.cos(CesiumMath.PI_OVER_TWO);
+        var s3PiOver2 = Math.sin(CesiumMath.PI_OVER_TWO * 3);
+        var c3PiOver2 = Math.cos(CesiumMath.PI_OVER_TWO * 3);
+
+        var headingPitchRoll = new HeadingPitchRoll(-CesiumMath.PI_OVER_TWO, -CesiumMath.PI, CesiumMath.PI_OVER_TWO * 3);
+        var expected = new Matrix3(cPi * cPiOver2, -c3PiOver2 * sPiOver2 + s3PiOver2 * sPi * cPiOver2, s3PiOver2 * sPiOver2 + c3PiOver2 * sPi * cPiOver2,
```

Comment:
Unfortunately I don't think this approach is good either.  It's still circular in that we're testing our own code against our own code.  There may even be a chance that under the hood one of these are written in terms of the other (though I didn't check).

This test is also poor because it uses PI/2 and other "nice" values that can result in simple cases working fine but complicated cases being wrong. (I think some of the terms of the matrix end up being 0 for example).

Math unit tests like this should always contain hard numbers that have been validated to be correct either manually or via other trusted sources.  I'll try and whip up a test case using STK Components.